File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let complements = {};
    for(let i = 0; i < nums.length; i++){
        let diff = target - nums[i];
        if(diff in complements){
            return [i, complements[diff]];
        }
        complements[nums[i]] = i;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement = {} #num to index of num
        
        for i in range(0, len(nums)):
            diff = target - nums[i]
            if diff in complement:
                return [i, complement[diff]]
            else:
                complement[nums[i]] = i
                    